provider/infoblox.go

```diff
@@ -0,0 +1,483 @@
+	}
+	for _, change := range changes.Create {
+		mapChange(created, change)
+	}
+	for _, change := range changes.UpdateNew {
+		mapChange(created, change)
+	}
+
+	return created, deleted, updated
+}
+
+func (p *InfobloxProvider) findZone(zones []ibclient.ZoneAuth, name string) *ibclient.ZoneAuth {
+	var result *ibclient.ZoneAuth
+
+	// Go through every zone looking for the longest name (i.e. most specific) as a matching suffix
+	for idx := range zones {
+		zone := &zones[idx]
+		if strings.HasSuffix(name, zone.Fqdn) {
+			if result == nil || len(zone.Fqdn) > len(result.Fqdn) {
+				result = zone
+			}
+		}
+	}
+	return result
+}
+
+func (p *InfobloxProvider) mapObjRes(ep *endpoint.Endpoint, getObject bool) (objRes ibObjRes, err error) {
+	var obj ibclient.IBObject
+	switch ep.RecordType {
+	case endpoint.RecordTypeA:
+		var res []ibclient.RecordA
+		obj = ibclient.NewRecordA(
+			ibclient.RecordA{
+				Name:     ep.DNSName,
+				Ipv4Addr: ep.Target,
+			},
+		)
+		if getObject {
+			err = p.client.GetObject(obj, "", &res)
+			if err != nil {
+				return
+			}
+		}
+		objRes = ibObjRes{
+			obj: obj,
+			res: &res,
+		}
+	case endpoint.RecordTypeCNAME:
+		var res []ibclient.RecordCNAME
+		obj = ibclient.NewRecordCNAME(
+			ibclient.RecordCNAME{
+				Name:      ep.DNSName,
+				Canonical: ep.Target,
+			},
+		)
+		if getObject {
+			err = p.client.GetObject(obj, "", &res)
+			if err != nil {
+				return
+			}
+		}
+		objRes = ibObjRes{
+			obj: obj,
+			res: &res,
+		}
+	case endpoint.RecordTypeTXT:
+		var res []ibclient.RecordTXT
+		// The Infoblox API strips enclosing double quotes from TXT records lacking whitespace.
+		// Here we reconcile that fact by making this state match that reality.
+		if target, err := strconv.Unquote(ep.Target); err == nil && !strings.Contains(ep.Target, " ") {
+			ep.Target = target
+		}
+		obj = ibclient.NewRecordTXT(
+			ibclient.RecordTXT{
+				Name: ep.DNSName,
+				Text: ep.Target,
+			},
+		)
+		if getObject {
+			err = p.client.GetObject(obj, "", &res)
+			if err != nil {
+				return
+			}
+		}
+		objRes = ibObjRes{
+			obj: obj,
+			res: &res,
+		}
+	}
+	return
+}
+
+func (p *InfobloxProvider) createRecords(created infobloxChangeMap) {
+	for zone, endpoints := range created {
+		for _, ep := range endpoints {
+			if p.dryRun {
+				logrus.Infof(
+					"Would create %s record named '%s' to '%s' for Infoblox DNS zone '%s'.",
+					ep.RecordType,
+					ep.DNSName,
+					ep.Target,
+					zone,
+				)
+				continue
+			}
+
+			logrus.Infof(
+				"Creating %s record named '%s' to '%s' for Infoblox DNS zone '%s'.",
+				ep.RecordType,
+				ep.DNSName,
+				ep.Target,
+				zone,
+			)
+
+			mapObjRes, err := p.mapObjRes(ep, false)
+			if err != nil {
+				logrus.Errorf(
+					"Failed to retrieve %s record named '%s' to '%s' for DNS zone '%s': %v",
+					ep.RecordType,
+					ep.DNSName,
+					ep.Target,
+					zone,
+					err,
+				)
+				continue
+			}
+			_, err = p.client.CreateObject(mapObjRes.obj)
```

Comment:
I guess, if mapObjRes (and please rename it) would have a method create() and a member client, this would be more readable

Reply:
I'm really not sure I understand the problem.